### PR TITLE
satellite/console: project members refactoring

### PR DESCRIPTION
**Pull request by Barterio**
What: 
   Project members queries and service refactored to have code consistency for all Paged requests
Why:
   
Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/console/consoleweb/consoleql/project.go [hunk 1]
```diff
@@ -421,3 +419,20 @@ func fromMapBucketUsageCursor(args map[string]interface{}) (cursor console.Bucke
 	cursor.Search, _ = args[SearchArg].(string)
 	return
 }
+
+func fromMapProjectMembersCursor(args map[string]interface{}) console.ProjectMembersCursor {
```
Comment (egonelbre) [inline satellite/console/consoleweb/consoleql/project.go]: `cursorArgsToProjectMembersCursor`

File: satellite/satellitedb/projectmembers.go [hunk 2]
```diff
@@ -51,11 +87,19 @@ func (pm *projectMembers) GetByProjectID(ctx context.Context, projectID uuid.UUI
 					AND ( u.email LIKE ? OR
 						  u.full_name LIKE ? OR
 						  u.short_name LIKE ? )
-						ORDER BY ` + sanitizedOrderColumnName(pagination.Order) + ` ASC
+						ORDER BY ` + sanitizedOrderColumnName(cursor.Order) + ` 
+						` + sanitizeOrderDirectionName(page.OrderDirection)+ `
```
Comment (rikysya) [inline satellite/satellitedb/projectmembers.go]: Direction for page is not set

File: satellite/console/consoleweb/consoleql/project.go [hunk 3]
```diff
@@ -99,48 +107,28 @@ func graphqlProject(service *console.Service, types *TypeCreator) *graphql.Objec
...
-					}
-
-					members, err := service.GetProjectMembers(p.Context, project.ID, pagination)
+					_, err := console.GetAuth(p.Context)
 					if err != nil {
 						return nil, err
 					}
 
-					_, err = console.GetAuth(p.Context)
+					inputPage := fromMapProjectMembersCursor(p.Args[CursorArg].(map[string]interface{}))
```
Comment (rikysya) [inline satellite/console/consoleweb/consoleql/project.go]: why Page?

File: satellite/satellitedb/projectmembers.go [hunk 4]
```diff
@@ -32,16 +32,52 @@ func (pm *projectMembers) GetByMemberID(ctx context.Context, memberID uuid.UUID)
...
+		return nil, errs.New("page cannot be 0")
+	}
+
+	page := &console.ProjectMembersPage{
+		Search: cursor.Search,
+		Limit:  cursor.Limit,
+		Offset: uint64((cursor.Page - 1) * cursor.Limit),
+	}
 
+	countQuery := pm.db.Rebind(`
```
Comment (egonelbre) [inline satellite/satellitedb/projectmembers.go]:
```
	countQuery := pm.db.Rebind(`
		SELECT COUNT(*)
		FROM project_members
		INNER JOIN users ON project_members.member_id = u.id
		WHERE project_members.project_id = ?
			AND ( users.email LIKE ? OR 
				  users.full_name LIKE ? OR
				  users.short_name LIKE ? 
			)`)
```

The `WHERE` belongs to the main `SELECT` so might make more sense to remove the identation.

File: satellite/satellitedb/projectmembers.go [hunk 5]
```diff
@@ -91,7 +136,17 @@ func (pm *projectMembers) GetByProjectID(ctx context.Context, projectID uuid.UUI
...
 	}
 
-	return projectMembers, err
+	page.ProjectMembers = projectMembers
+	page.Order = cursor.Order
+
+	page.PageCount = uint(page.TotalCount / uint64(cursor.Limit))
+	if page.TotalCount % uint64(cursor.Limit) != 0 {
+		page.PageCount++
+	}
```
Comment (egonelbre) [inline satellite/satellitedb/projectmembers.go]: Isn't this the same as `page.PageCount = (page.TotalCount - 1) / cursor.Limit`

**Pull request merged**